src/main/java/com/example/springbootbookshop/repository/UserRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.example.springbootbookshop.repository;
+
+import com.example.springbootbookshop.entity.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+
+    Optional<User> findByEmail(String email);
```

Comment:
You can add `@EntityGraph(attributePaths = "roles")` to avoid declare Eager fetch type in your entity